x-pack/platform/plugins/private/indices_metadata/server/lib/services/indices_metadata.ts

```diff
@@ -0,0 +1,301 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Subscription } from 'rxjs';
+import type {
+  ConcreteTaskInstance,
+  TaskInstance,
+  TaskManagerSetupContract,
+  TaskManagerStartContract,
+} from '@kbn/task-manager-plugin/server';
+import type { AnalyticsServiceStart, ElasticsearchClient, LogMeta, Logger } from '@kbn/core/server';
+import type {
+  DataStream,
+  DataStreams,
+  IlmPolicies,
+  IlmsStats,
+  IndexSettings,
+  IndexTemplateInfo,
+  IndexTemplatesStats,
+  IndicesMetadataConfiguration,
+  IndicesSettings,
+  IndicesStats,
+} from './indices_metadata.types';
+
+import {
+  DATA_STREAM_EVENT,
+  ILM_POLICY_EVENT,
+  ILM_STATS_EVENT,
+  INDEX_STATS_EVENT,
+  INDEX_SETTINGS_EVENT,
+  INDEX_TEMPLATES_EVENT,
+} from '../ebt/events';
+import { MetadataReceiver } from './receiver';
+import { MetadataSender } from './sender';
+import { ConfigurationService } from './configuration';
+
+const TASK_TYPE = 'IndicesMetadata:IndicesMetadataTask';
+const TASK_ID = 'indices-metadata:indices-metadata-task:1.0.0';
+const INTERVAL = '1m';
```

Comment:
How/when will this interval be changed and what is the plan for getting this into production? I'd also be curious to get an idea of how much data this potentially pushes.